R/methSeg.R

```diff
@@ -1,5 +1,110 @@
 # segmentation functions 
 
+.run.fastseg = function(obj, ...){
+  
+  dots <- list(...)
+  
+  ## check wether obj contains at least one metacol 
+  if(ncol(elementMetadata(obj))<1)
+    stop("GRanges does not have any meta column.")
+  
+  ## check wether obj contains is sorted by position
+  if(is.unsorted(obj,ignore.strand=TRUE)) {
+    obj <- sort(obj,ignore.strand=TRUE)
+    message("Object not sorted by position, sorting now.")
+  }
+  
+  ## check wether obj contains at least two ranges else stop
+  if(length(obj)<=1)
+    stop("segmentation requires at least two ranges.")
+  
+  # match argument names to fastseg arguments
+  args.fastseg=dots[names(dots) %in% names(formals(fastseg)[-1] ) ]  
+  
+  args.fastseg[["x"]]=obj
+  
+  # do the segmentation
+  #seg.res=fastseg(obj)
+  seg.res <- do.call("fastseg", args.fastseg)
+  
+  # stop if segmentation produced only one range
+  if(length(seg.res)==1) {
```

Comment:
yes, I think I know what you mean, this if clause is here in the original code and you return seg.res if there is only 1 segment https://github.com/al2na/methylKit/blob/0d007f20b203cc567e90bc40227534007c95034b/R/methSeg.R#L127 but since I separated lines fo code for calling mclust into a auxiliary function, I need to make sure that this function ( .run.mclust() ) won't be run if there is only 1 segment..